packages/node-provider/src/node-provider.ts

```diff
@@ -1,7 +1,66 @@
-import { INodeProvider, NodeMessage } from "./types";
+import EventEmitter from "eventemitter3";
+
+import {
+  INodeProvider,
+  NodeMessage,
+  NodeMessageReceivedCallback
+} from "./types";
```

Comment:
I think these types should be moved to cfjs/types. If not, then CFjs becomes dependent on the NodeProvider package, which is weird. Plus, these types correspond to the written spec inside CFjs.